datadog/resource_datadog_role.go

```diff
@@ -0,0 +1,287 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+// validPermissions is a map of all unrestricted permission IDs to their name
+var validPermissions map[string]string
+
+func resourceDatadogRole() *schema.Resource {
+	return &schema.Resource{
+		Exists: resourceDatadogRoleExists,
+		Create: resourceDatadogRoleCreate,
+		Read:   resourceDatadogRoleRead,
+		Update: resourceDatadogRoleUpdate,
+		Delete: resourceDatadogRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogRoleImport,
+		},
+		CustomizeDiff: customdiff.ValidateValue("permission", validatePermissionsUnrestricted),
+		Schema: map[string]*schema.Schema{
```

Comment:
should this include the role UUID?

Reply:
ID's are specific in terraform. They are not in the schema, but they are set in the state (look for `SetId()`) and user can reference it in other resources with `resource_name.id`